modules/apps/segments/segments-experiment-web/src/main/java/com/liferay/segments/experiment/web/internal/servlet/taglib/SegmentsExperimentAnalyticsTopHeadDynamicInclude.java

```diff
@@ -0,0 +1,134 @@
+
+import java.io.IOException;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Eduardo García
+ */
+@Component(immediate = true, service = DynamicInclude.class)
+public class SegmentsExperimentAnalyticsTopHeadDynamicInclude
+	implements DynamicInclude {
+
+	@Override
+	public void include(
+			HttpServletRequest httpServletRequest,
+			HttpServletResponse httpServletResponse, String key)
+		throws IOException {
+
+		ThemeDisplay themeDisplay =
+			(ThemeDisplay)httpServletRequest.getAttribute(
+				WebKeys.THEME_DISPLAY);
+
+		if (!SegmentsExperimentUtil.isAnalyticsEnabled(
+				themeDisplay.getCompanyId())) {
+
+			return;
+		}
+
+		SegmentsExperiment segmentsExperiment =
+			(SegmentsExperiment)httpServletRequest.getAttribute(
+				SegmentsExperimentWebKeys.SEGMENTS_EXPERIMENT);
+
+		long[] segmentsExperienceIds = GetterUtil.getLongValues(
+			httpServletRequest.getAttribute(
+				SegmentsWebKeys.SEGMENTS_EXPERIENCE_IDS));
+
+		StringBundler sb = StringUtil.replaceToStringBundler(
+			_ANALYTICS_TMPL_CONTENT, "${", "}",
+			_getAnalyticsContextMap(
+				segmentsExperiment,
+				_getSegmentsExperienceKey(segmentsExperienceIds)));
+
+		sb.writeTo(httpServletResponse.getWriter());
+	}
+
+	@Override
+	public void register(DynamicIncludeRegistry dynamicIncludeRegistry) {
+		dynamicIncludeRegistry.register(
+			"/dynamic_include/top_head.jsp#analytics");
+	}
+
+	private Map<String, String> _getAnalyticsContextMap(
+		SegmentsExperiment segmentsExperiment,
+		String segmentsExperimentSegmentsExperienceKey) {
+
+		Map<String, String> replaceMap = new HashMap<>();
+
+		if (segmentsExperiment == null) {
+			replaceMap.put(
+				"experienceId", segmentsExperimentSegmentsExperienceKey);
+			replaceMap.put("experimentId", StringPool.BLANK);
+			replaceMap.put("variantId", StringPool.BLANK);
+
+			return replaceMap;
+		}
+
+		replaceMap.put(
+			"experienceId", segmentsExperiment.getSegmentsExperienceKey());
+		replaceMap.put(
+			"experimentId", segmentsExperiment.getSegmentsExperimentKey());
+		replaceMap.put("variantId", segmentsExperimentSegmentsExperienceKey);
+
+		return replaceMap;
+	}
+
+	private String _getSegmentsExperienceKey(long[] segmentsExperienceIds) {
+		if (segmentsExperienceIds.length > 0) {
+			SegmentsExperience segmentsExperience =
+				_segmentsExperienceLocalService.fetchSegmentsExperience(
+					segmentsExperienceIds[0]);
+
+			if (segmentsExperience != null) {
+				return segmentsExperience.getSegmentsExperienceKey();
+			}
+		}
+
+		return SegmentsExperienceConstants.KEY_DEFAULT;
+	}
+
+	private static final String _ANALYTICS_TMPL_CONTENT = StringUtil.read(
+		SegmentsExperimentAnalyticsTopHeadJSPDynamicInclude.class,
+		"analytics.tmpl");
```

Comment:
Maybe you already noticed but the name here doesn't match the "analytics.tpl" resource.